### PR TITLE
Gradle plugin refactorings

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt [hunk 1]
```diff
@@ -173,6 +170,9 @@ internal class OrtModelBuilder : ToolingModelBuilder {
         }.filterIsInstance<ResolvedArtifactResult>()
     }
 
+    private fun AttributeContainer.getValueByName(name: String): Any? =
```
Comment (fviernau) [inline plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt:173]: Could be move out of class.

File: plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt [hunk 2]
```diff
@@ -173,6 +170,9 @@ internal class OrtModelBuilder : ToolingModelBuilder {
...
+        attributes.keySet().find { it.name == name }?.let { getAttribute(it) }
+
```
Comment (fviernau) [inline plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt:175]: Is it not possible to do `getAttribute<Any>("key")`?

Comment (sschuberth) [reply plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt:175]: No, as the key into the container is the attribute (which consist of a name and a class type), but just the name. Which means you usually need to know the type of the attribute before you can look up its value. The whole point of the function is to be able to look up an attribute's value irrespective of the type, just by its name, as there sometimes are inconsistencies in the type an attribute uses (see the deleted comment, which is now not relevant anymore).

File: plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt (hunk 1 above)
Comment (sschuberth) [reply plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt]: I've added a commit that introduces a new file to bundle these extension functions.

**Review by fviernau — APPROVED** [review]

**Pull request merged**